arpeggio/tests/test_error_reporting.py

```diff
@@ -119,11 +163,18 @@ def grammar():
 
     with pytest.raises(NoMatch) as e:
         parser.parse('   one ident')
-    assert "Not expected input" in str(e.value)
+    assert (
+        "Expected (NOT('one') AND /\\w+/) at position (1, 1) => '*   one ide'."
+        == str(e.value)
+    )
 
 
 def test_not_match_as_alternative():
     """
+    # ##########################################################################
+    # WIP: We have doubts about this behavior. To our understanding,
+    # the parsing of OrderedChoice should not throw.
+    # ##########################################################################
```

Comment:
You are right. There has already been a related discussion in #96. The issue also influence #98. I started working it.

Reply:
Thanks for the contribution  @mettta and @stanislaw. I think you are heading in the right direction but there is an issue that you have noticed (and was reported recently in #96) that needs to be handled before your work is integrated. I'm working on it and will do a detailed review of this PR when finished.

Reply:
Hi, I just finished the work I've mentioned. Please rebase both your PRs on the `master` to see if everything works.

Reply:
We will rebase in just a few minutes. We are preparing one small commit on top of this patch that makes all previous tests pass as they were in the master branch (reporting only the non-matched expressions in a non-matched expression, not the whole expression). We are fixing the remaining tests and don't know yet how long this should take.

Reply:
Excellent. Thanks a lot! Just take your time. Nothing urgent here :)